Sources/PostgresKit/PostgresDataDecoder.swift

```diff
@@ -140,7 +143,11 @@ public final class PostgresDataDecoder {
 
         func decode<T>(_ type: T.Type) throws -> T where T : Decodable {
             if let convertible = T.self as? PostgresDataConvertible.Type {
-                return convertible.init(postgresData: self.data)! as! T
+                guard let converted = convertible.init(postgresData: self.data) as? T else {
+                    throw Error.valueNotFound
```

Comment:
Throwing a [`DecodingError`](https://developer.apple.com/documentation/swift/decodingerror) might be better here since that's what the other decoders do in similar situations. That has a `valueNotFound` case.